Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -16,6 +21,178 @@
+                // imageBoundsY)
+                AZ::Vector3 uvw(static_cast<float>(x), static_cast<float>((imageResolutionY - 1) - y), 0.0f);
+
+                GradientSampleParams sampleParams;
+                sampleParams.m_position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
+
+                bool inBounds = true;
+                LmbrCentral::ShapeComponentRequestsBus::EventResult(
+                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, sampleParams.m_position);
+
+                float sample = inBounds ? m_configuration.m_gradientSampler.GetValue(sampleParams) : 0.0f;
+
+                // Write out the sample value for the pixel based on output format
+                int index = ((y * imageResolutionX) + x) * channels;
+                switch (m_configuration.m_outputFormat)
+                {
+                case OutputFormat::R8:
+                {
+                    AZ::u8 value = static_cast<AZ::u8>(sample * std::numeric_limits<AZ::u8>::max());
+                    pixels[index] = value; // R
+
+                    if (channels == 4)
+                    {
+                        pixels[index + 1] = value; // G
+                        pixels[index + 2] = value; // B
+                        pixels[index + 3] = std::numeric_limits<AZ::u8>::max(); // A
+                    }
+                    break;
+                }
+                case OutputFormat::R16:
+                {
+                    auto actualMem = reinterpret_cast<AZ::u16*>(pixels.data());
+                    AZ::u16 value = static_cast<AZ::u16>(sample * std::numeric_limits<AZ::u16>::max());
+                    actualMem[index] = value; // R
+
+                    if (channels == 4)
+                    {
+                        actualMem[index + 1] = value; // G
+                        actualMem[index + 2] = value; // B
+                        actualMem[index + 3] = std::numeric_limits<AZ::u16>::max(); // A
+                    }
+                    break;
+                }
+                case OutputFormat::R32:
+                {
+                    auto actualMem = reinterpret_cast<float*>(pixels.data());
+                    actualMem[index] = sample; // R
+
+                    if (channels == 4)
+                    {
+                        actualMem[index + 1] = sample; // G
+                        actualMem[index + 2] = sample; // B
+                        actualMem[index + 3] = 1.0f; // A
+                    }
+                    break;
+                }
+                }
+            }
+        }
+
+        bool result = outputImage->write_image(pixelFormat, pixels.data());
+        if (!result)
+        {
+            AZ_Error("GradientBaker", false, "Failed to write out gradient baked image to path: %s",
+                absolutePath.c_str());
+        }
```

Comment:
Nit: We can replace the constant false with `result` in the error message. Also need a maybe_unused at declaration for release optimization.